pkg/utils/types.go

```diff
@@ -6,10 +6,18 @@ import (
 )
 
 type Validator struct {
+	Name        string
```

Comment:
Why is this attribute needed? The `AllValidators` seems to track it via it's key right and at the same time, ensures uniqueness of them :)

Reply:
If you are familiar with Hadolint and ShellCheck, #40 is the first step towards a similar approach.

The idea is, there is no `failureMsg` good enough to fully explain a static check failure, so we will instead populate a wiki: https://github.com/mt-sre/addon-metadata-operator/wiki/Home, with 1 page per validator.

Upon failure, we will throw a standardized failureMsg, that points to the wiki. The `Name` attribute can be like pylint, where you have a code `AMXXXX` that doesn't mean anything, but the name is more human friendly.

```go
func (v Validator) Fail(failureMsg string) (bool, string, error) {
  msg := fmt.Sprintf("%v: (%v) %v (%v)", v.Code, v.Name, failureMsg, v.GetWikiURL)
  return false, msg, nil
}
```

WDYT?

Reply:
Got it! Really cool stuff.